extension/src/shared/view/selection/ColorSchemeSectionForComponentField.tsx

```diff
@@ -122,9 +128,26 @@ export const ColorSchemeSectionForComponentField: FC<ColorSchemeSectionForCompon
               isConditionalColorSchemeComponent(componentValue) ||
               isGradientColorSchemeComponent(componentValue)
             ) {
-              const ruleId = componentValue.value?.useDefault
-                ? componentValue.value?.currentRuleId ?? componentValue.preset?.rules?.[0]?.id
-                : componentValue.value?.currentRuleId;
+              const ruleId =
+                componentValue.value?.useDefault ||
+                componentValue.preset?.rules?.some(r => r.asDefaultRule)
+                  ? componentValue.value?.currentRuleId ??
+                    (
+                      (componentValue.preset?.rules as unknown[] | undefined)?.find(
+                        r =>
+                          (
+                            r as
+                              | FillColorConditionFieldPresetRule
+                              | FillAndStrokeColorConditionFieldPresetRule
+                              | FillColorGradientFieldPresetRule
+                          ).asDefaultRule,
+                      ) as
+                        | FillColorConditionFieldPresetRule
+                        | FillAndStrokeColorConditionFieldPresetRule
+                        | FillColorGradientFieldPresetRule
```

Comment:
I have to add these type `as` to avoid the type error when run `tsc`, but the type is the same without these `as` and it will not show type error in editor.